tests/vm_startup_test.go

```diff
@@ -0,0 +1,162 @@
+const dnsPort = "31111"
+const dnsIP = "127.0.0.1" // Forwarded to the node port - https://github.com/kubevirt/kubevirtci/pull/867
+const domain = "vm.secondary.io"
+
+var _ = Describe("Virtual Machines Startup", func() {
+
+	BeforeEach(func() {
+		By("Creating test namespace")
+		ns := k8sv1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testNamespace,
+			},
+		}
+		err := getClient().Create(context.Background(), &ns)
+		Expect(err).ToNot(HaveOccurred(), "couldn't create namespace for the test")
+	})
+
+	AfterEach(func() {
+		By("Removing all Virtual Machine Instances from the namespace")
+		vmiList := &v1.VirtualMachineInstanceList{}
+		err := getClient().List(context.Background(), vmiList, &client.ListOptions{Namespace: testNamespace})
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, vmiObject := range vmiList.Items {
+			err = getClient().Delete(context.Background(), &vmiObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Eventually(func() int {
+			vmiList := &v1.VirtualMachineInstanceList{}
+			err := getClient().List(context.Background(), vmiList, &client.ListOptions{Namespace: testNamespace})
+			Expect(err).ToNot(HaveOccurred())
+			return len(vmiList.Items)
+
+		}, timeout, pollingInterval).Should(Equal(0), "failed to remove all vm objects")
+	})
+
+	AfterEach(func() {
+		By("Removing all Network AttachmentDefinitions from the namespace")
+		nadList := &networkv1.NetworkAttachmentDefinitionList{}
+		err := getClient().List(context.Background(), nadList, &client.ListOptions{Namespace: testNamespace})
+		Expect(err).ToNot(HaveOccurred())
+
+		for _, nadObject := range nadList.Items {
+			err = getClient().Delete(context.Background(), &nadObject)
+			Expect(err).ToNot(HaveOccurred())
+		}
+
+		Eventually(func() int {
+			nadList := &networkv1.NetworkAttachmentDefinitionList{}
+			err := getClient().List(context.Background(), nadList, &client.ListOptions{Namespace: testNamespace})
+			Expect(err).ToNot(HaveOccurred())
+			return len(nadList.Items)
+
+		}, timeout, pollingInterval).Should(Equal(0), "failed to remove all nad objects")
+	})
+
+	AfterEach(func() {
+		By("Removing the test namespace")
+		namespace := &k8sv1.Namespace{}
+		err := getClient().Get(context.Background(), types.NamespacedName{Name: testNamespace}, namespace)
+		Expect(err).ToNot(HaveOccurred())
+		Eventually(func() error {
+			return getClient().Delete(context.Background(), namespace)
+		}, 240*time.Second, 1*time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), fmt.Sprintf("should successfully delete namespace '%s'", testNamespace))
+	})
+
+	Context("with one secondary and no default interface", func() {
+		It("should get the correct IP when nslookup to the secondary interface FQDN", func() {
+			By("Creating a NetworkAttachmentDefinition")
+			err := createNetworkAttachmentDefinition(testNamespace, "ptp-conf")
+			Expect(err).ToNot(HaveOccurred(), "Should successfully create the NAD")
```

Comment:
can be done please as a BeforeEach of this It (in the same Context), as it is not part of the "It" but preparation,
makes the It smaller

Reply:
done.